chunkset_tpl.h

```diff
@@ -34,37 +63,18 @@ Z_INTERNAL uint8_t* CHUNKCOPY(uint8_t *out, uint8_t const *from, unsigned len) {
     }
     return out;
 }
+Z_INTERNAL uint8_t* CSUFFIX(chunkcopy)(uint8_t *out, uint8_t const *from, unsigned len) {
+    return CSUFFIX(chunkcopy_static)(out, from, len);
+}
 
 /* Behave like chunkcopy, but avoid writing beyond of legal output. */
-Z_INTERNAL uint8_t* CHUNKCOPY_SAFE(uint8_t *out, uint8_t const *from, unsigned len, uint8_t *safe) {
-    if ((safe - out) < (ptrdiff_t)sizeof(chunk_t)) {
-        int32_t use_chunk16 = sizeof(chunk_t) > 16 && (len & 16);
-        if (use_chunk16) {
-            memcpy(out, from, 16);
-            out += 16;
-            from += 16;
-        }
-        if (len & 8) {
-            memcpy(out, from, 8);
-            out += 8;
-            from += 8;
-        }
-        if (len & 4) {
-            memcpy(out, from, 4);
-            out += 4;
-            from += 4;
-        }
-        if (len & 2) {
-            memcpy(out, from, 2);
-            out += 2;
-            from += 2;
-        }
-        if (len & 1) {
-            *out++ = *from++;
-        }
-        return out;
-    }
-    return CHUNKCOPY(out, from, len);
+static inline uint8_t* CSUFFIX(chunkcopy_safe_static)(uint8_t *out, uint8_t const *from, unsigned len, uint8_t *safe) {
+    if ((safe - out) < (ptrdiff_t)sizeof(chunk_t))
```

Comment:
I still think this line is bad. Taking distance of two pointers is never going to give correct length as the range is supposed to be inclusive on both start and end, so `safe` points to last valid memory location, not first out-of-bound location.

Reply:
Not sure if I need to do anything here.

Reply:
I have incorporated some of your changes.